PlugManX v3 reload error on Folia

Seems not using folia scheduler.
```
[01:24:04 WARN]: Unexpected exception while parsing console command "plugman reload esu"                                                                                                           
org.bukkit.command.CommandException: Unhandled exception executing 'plugman reload esu' in paper.com.rylinaux.plugman.commands.OldPaperCommandCreator$1(plugman)                                   
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:175) ~[server.jar:?]                                                                          
        at org.bukkit.craftbukkit.CraftServer.dispatchCommand(CraftServer.java:1044) ~[server-1.21.1.jar:1.21.1-DEV-f2c9dc9]                                                                         
        at org.bukkit.craftbukkit.CraftServer.dispatchServerCommand(CraftServer.java:995) ~[server-1.21.1.jar:1.21.1-DEV-f2c9dc9]                                                                    
        at net.minecraft.server.dedicated.DedicatedServer.handleConsoleInputs(DedicatedServer.java:533) ~[server-1.21.1.jar:1.21.1-DEV-f2c9dc9]                                                      
        at io.papermc.paper.threadedregions.RegionizedServer.globalTick(RegionizedServer.java:296) ~[server-1.21.1.jar:1.21.1-DEV-f2c9dc9]                                                           
        at io.papermc.paper.threadedregions.RegionizedServer$GlobalTickTickHandle.tickRegion(RegionizedServer.java:148) ~[server-1.21.1.jar:1.21.1-DEV-f2c9dc9]                                      
        at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:459) ~[server-1.21.1.jar:1.21.1-DEV-f2c9dc9]                                   
        at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[server-1.21.1.jar:1.21.1-DEV-f2c9dc9]                                    
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]                                                                                                                                 
Caused by: java.lang.UnsupportedOperationException                                                                                                                                                 
        at org.bukkit.craftbukkit.scheduler.CraftScheduler.handle(CraftScheduler.java:524) ~[server-1.21.1.jar:1.21.1-DEV-f2c9dc9]                                                                   
        at org.bukkit.craftbukkit.scheduler.CraftScheduler.runTaskTimer(CraftScheduler.java:228) ~[server-1.21.1.jar:1.21.1-DEV-f2c9dc9]                                                             
        at org.bukkit.craftbukkit.scheduler.CraftScheduler.runTaskTimer(CraftScheduler.java:210) ~[server-1.21.1.jar:1.21.1-DEV-f2c9dc9]                                                             
        at org.bukkit.craftbukkit.scheduler.CraftScheduler.runTaskLater(CraftScheduler.java:174) ~[server-1.21.1.jar:1.21.1-DEV-f2c9dc9]                                                             
        at PlugManX-3.0.3-RC.1.jar/bukkit.com.rylinaux.plugman.pluginmanager.BasePluginManager.scheduleCommandLoading(BasePluginManager.java:160) ~[PlugManX-3.0.3-RC.1.jar:?]                     
        at PlugManX-3.0.3-RC.1.jar/bukkit.com.rylinaux.plugman.pluginmanager.BukkitPluginManager.load(BukkitPluginManager.java:361) ~[PlugManX-3.0.3-RC.1.jar:?]                                   
        at PlugManX-3.0.3-RC.1.jar/core.com.rylinaux.plugman.plugins.PluginManager.load(PluginManager.java:173) ~[PlugManX-3.0.3-RC.1.jar:?]                                                       
        at PlugManX-3.0.3-RC.1.jar/paper.com.rylinaux.plugman.pluginmanager.PaperPluginManager.load(PaperPluginManager.java:59) ~[PlugManX-3.0.3-RC.1.jar:?]                                       
        at PlugManX-3.0.3-RC.1.jar/core.com.rylinaux.plugman.commands.executables.ReloadCommand.reloadPlugin(ReloadCommand.java:137) ~[PlugManX-3.0.3-RC.1.jar:?]
        at PlugManX-3.0.3-RC.1.jar/core.com.rylinaux.plugman.commands.executables.ReloadCommand.execute(ReloadCommand.java:100) ~[PlugManX-3.0.3-RC.1.jar:?]
        at PlugManX-3.0.3-RC.1.jar/bukkit.com.rylinaux.plugman.commands.PlugManCommandHandler.onCommand(PlugManCommandHandler.java:70) ~[PlugManX-3.0.3-RC.1.jar:?]
        at PlugManX-3.0.3-RC.1.jar/paper.com.rylinaux.plugman.commands.OldPaperCommandCreator$1.execute(OldPaperCommandCreator.java:20) ~[PlugManX-3.0.3-RC.1.jar:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:165) ~[server.jar:?]
        ... 8 more
```

--- a/plugman-assembly/pom.xml
+++ b/plugman-assembly/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.3</version>
+        <version>3.0.4-dev1</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-assembly/pom.xml
+++ b/plugman-assembly/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.4-dev1</version>
+        <version>3.0.4</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-bukkit/pom.xml
+++ b/plugman-bukkit/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.3</version>
+        <version>3.0.4-dev1</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-bukkit/pom.xml
+++ b/plugman-bukkit/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.4-dev1</version>
+        <version>3.0.4</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-bukkit/src/main/java/bukkit/com/rylinaux/plugman/pluginmanager/BasePluginManager.java
+++ b/plugman-bukkit/src/main/java/bukkit/com/rylinaux/plugman/pluginmanager/BasePluginManager.java
@@ -33,8 +33,8 @@ import core.com.rylinaux.plugman.plugins.Command;
 import core.com.rylinaux.plugman.plugins.CommandMapWrap;
 import core.com.rylinaux.plugman.plugins.Plugin;
 import core.com.rylinaux.plugman.plugins.PluginManager;
+import core.com.rylinaux.plugman.util.ThreadUtil;
 import core.com.rylinaux.plugman.util.reflection.FieldAccessor;
-import org.bukkit.Bukkit;
 import org.bukkit.command.PluginCommand;
 import org.bukkit.command.SimpleCommandMap;
 import org.bukkit.event.Event;
@@ -157,7 +157,7 @@ public abstract class BasePluginManager implements PluginManager {
      * Common command loading logic.
      */
     protected synchronized void scheduleCommandLoading() {
-        Bukkit.getScheduler().runTaskLater(PlugManBukkit.getInstance(), this::syncCommands, 10L);
+        PlugManBukkit.getInstance().get(ThreadUtil.class).syncLater(this::syncCommands, 10L);
     }
 
 

--- a/plugman-bukkit/src/main/java/bukkit/com/rylinaux/plugman/util/BukkitThreadUtil.java
+++ b/plugman-bukkit/src/main/java/bukkit/com/rylinaux/plugman/util/BukkitThreadUtil.java
@@ -44,7 +44,7 @@ public class BukkitThreadUtil implements ThreadUtil {
      */
     @Override
     public void async(Runnable runnable) {
-        Bukkit.getScheduler().runTaskAsynchronously(PlugManBukkit.getInstance(), runnable);
+        asyncLater(runnable, 0);
     }
 
     /**
@@ -54,7 +54,17 @@ public class BukkitThreadUtil implements ThreadUtil {
      */
     @Override
     public void sync(Runnable runnable) {
-        Bukkit.getScheduler().runTask(PlugManBukkit.getInstance(), runnable);
+        syncLater(runnable, 0);
+    }
+
+    @Override
+    public void asyncLater(Runnable runnable, long delay) {
+        Bukkit.getScheduler().runTaskLaterAsynchronously(PlugManBukkit.getInstance(), runnable, delay);
+    }
+
+    @Override
+    public void syncLater(Runnable runnable, long delay) {
+        Bukkit.getScheduler().runTaskLater(PlugManBukkit.getInstance(), runnable, delay);
     }
 
     @Override

--- a/plugman-bungee/pom.xml
+++ b/plugman-bungee/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.3</version>
+        <version>3.0.4-dev1</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-bungee/pom.xml
+++ b/plugman-bungee/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.4-dev1</version>
+        <version>3.0.4</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-bungee/src/main/java/bungee/com/rylinaux/plugman/util/BungeeThreadUtil.java
+++ b/plugman-bungee/src/main/java/bungee/com/rylinaux/plugman/util/BungeeThreadUtil.java
@@ -10,12 +10,31 @@ public class BungeeThreadUtil implements ThreadUtil {
 
     @Override
     public void async(Runnable runnable) {
-        ProxyServer.getInstance().getScheduler().runAsync(PlugManBungee.getInstance(), runnable);
+        asyncLater(runnable, 0);
     }
 
     @Override
     public void sync(Runnable runnable) {
-        ProxyServer.getInstance().getScheduler().schedule(PlugManBungee.getInstance(), runnable, 0L, TimeUnit.MILLISECONDS);
+        syncLater(runnable, 0);
+    }
+
+    @Override
+    public void syncLater(Runnable runnable, long delay) {
+        ProxyServer.getInstance().getScheduler().schedule(PlugManBungee.getInstance(), runnable, delay / 20, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void asyncLater(Runnable runnable, long delay) {
+        Runnable tempRunnable = () -> {
+            ProxyServer.getInstance().getScheduler().runAsync(PlugManBungee.getInstance(), runnable);
+        };
+
+        if (delay <= 0) {
+            tempRunnable.run();
+            return;
+        }
+
+        syncLater(tempRunnable, delay);
     }
 
     @Override

--- a/plugman-core/pom.xml
+++ b/plugman-core/pom.xml
@@ -8,7 +8,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.3</version>
+        <version>3.0.4-dev1</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-core/pom.xml
+++ b/plugman-core/pom.xml
@@ -8,7 +8,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.4-dev1</version>
+        <version>3.0.4</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-core/src/main/java/core/com/rylinaux/plugman/commands/executables/DumpCommand.java
+++ b/plugman-core/src/main/java/core/com/rylinaux/plugman/commands/executables/DumpCommand.java
@@ -36,6 +36,7 @@ import java.io.File;
 import java.io.IOException;
 import java.io.PrintWriter;
 import java.nio.file.Path;
+import java.util.List;
 
 /**
  * Command that dumps plugin names and versions to file.
@@ -88,15 +89,15 @@ public class DumpCommand extends AbstractCommand {
     public void execute(CommandSender sender, String label, String[] args) {
         var dumpFile = new File(Path.of("plugins", "PlugManX").toFile(), "versions.txt");
 
-        var plugins = getPluginManager().getPluginNames(true);
-        plugins.sort(String.CASE_INSENSITIVE_ORDER);
+        var enabledPlugins = getPluginManager().getEnabledPluginNames(true);
+        enabledPlugins.sort(String.CASE_INSENSITIVE_ORDER);
+
+        var disabledPlugins = getPluginManager().getDisabledPluginNames(true);
+        disabledPlugins.sort(String.CASE_INSENSITIVE_ORDER);
 
         try (var writer = new PrintWriter(dumpFile)) {
-            for (var plugin : plugins) {
-                plugin = reformat(plugin);
-
-                writer.println(plugin);
-            }
+            write(enabledPlugins, writer, true);
+            write(disabledPlugins, writer, false);
             writer.flush();
 
             sender.sendMessage("dump.dumped", dumpFile.getName());
@@ -107,13 +108,20 @@ public class DumpCommand extends AbstractCommand {
         }
     }
 
-    private String reformat(String plugin) {
-        if (plugin.startsWith("§a")) plugin = plugin + " - Enabled";
+    private String reformat(String plugin, boolean enabled) {
+        if (enabled) plugin = plugin + " - Enabled";
+        else plugin = plugin + " - Disabled";
 
-        if (plugin.startsWith("§c")) plugin = plugin + " - Disabled";
-
-        plugin = plugin.substring(2);
+        if (plugin.startsWith("§")) plugin = plugin.substring(2);
 
         return plugin;
+    }
+
+    private void write(List<String> plugins, PrintWriter writer, boolean enabled) {
+        for (var plugin : plugins) {
+            plugin = reformat(plugin, enabled);
+
+            writer.println(plugin);
+        }
     }
 }

--- a/plugman-core/src/main/java/core/com/rylinaux/plugman/util/ThreadUtil.java
+++ b/plugman-core/src/main/java/core/com/rylinaux/plugman/util/ThreadUtil.java
@@ -5,6 +5,10 @@ public interface ThreadUtil {
 
     void sync(Runnable runnable);
 
+    void asyncLater(Runnable runnable, long delay);
+
+    void syncLater(Runnable runnable, long delay);
+
     void syncRepeating(Runnable runnable, long delay, long period);
 
     void asyncRepeating(Runnable runnable, long delay, long period);

--- a/plugman-paper/plugman-paper.iml
+++ b/plugman-paper/plugman-paper.iml
@@ -10,4 +10,7 @@
       </configuration>
     </facet>
   </component>
+  <component name="McpModuleSettings">
+    <option name="srgType" value="SRG" />
+  </component>
 </module>

--- a/plugman-paper/pom.xml
+++ b/plugman-paper/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.3</version>
+        <version>3.0.4-dev1</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-paper/pom.xml
+++ b/plugman-paper/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.4-dev1</version>
+        <version>3.0.4</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-paper/src/main/java/paper/com/rylinaux/plugman/util/PaperThreadUtil.java
+++ b/plugman-paper/src/main/java/paper/com/rylinaux/plugman/util/PaperThreadUtil.java
@@ -65,7 +65,8 @@ public class PaperThreadUtil extends BukkitThreadUtil {
     /**
      * Run a task in the main thread with Folia support.
      */
-    synchronized public void syncLater(Runnable runnable, long delay) {
+    @Override
+    public synchronized void syncLater(Runnable runnable, long delay) {
         if (!shouldUseFolia()) {
             super.syncLater(runnable, delay);
             return;

--- a/plugman-paper/src/main/java/paper/com/rylinaux/plugman/util/PaperThreadUtil.java
+++ b/plugman-paper/src/main/java/paper/com/rylinaux/plugman/util/PaperThreadUtil.java
@@ -52,28 +52,27 @@ public class PaperThreadUtil extends BukkitThreadUtil {
      * Run a task in a separate thread with Folia support.
      */
     @Override
-    public void async(Runnable runnable) {
+    public void asyncLater(Runnable runnable, long delay) {
         if (!shouldUseFolia()) {
             super.async(runnable);
             return;
         }
 
         var foliaLib = new com.tcoded.folialib.FoliaLib(PlugManBukkit.getInstance());
-        foliaLib.getScheduler().runAsync((ignored) -> runnable.run());
+        foliaLib.getScheduler().runLaterAsync((ignored) -> runnable.run(), delay);
     }
 
     /**
      * Run a task in the main thread with Folia support.
      */
-    @Override
-    public void sync(Runnable runnable) {
+    synchronized public void syncLater(Runnable runnable, long delay) {
         if (!shouldUseFolia()) {
-            super.sync(runnable);
+            super.syncLater(runnable, delay);
             return;
         }
 
         var foliaLib = new com.tcoded.folialib.FoliaLib(PlugManBukkit.getInstance());
-        foliaLib.getScheduler().runLater(runnable, 0L);
+        foliaLib.getScheduler().runLater(runnable, delay);
     }
 
     /**

--- a/plugman-velocity/pom.xml
+++ b/plugman-velocity/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.3</version>
+        <version>3.0.4-dev1</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-velocity/pom.xml
+++ b/plugman-velocity/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>com.rylinaux</groupId>
         <artifactId>PlugManX</artifactId>
-        <version>3.0.4-dev1</version>
+        <version>3.0.4</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/plugman-velocity/src/main/java/velocity/com/rylinaux/plugman/PlugManVelocity.java
+++ b/plugman-velocity/src/main/java/velocity/com/rylinaux/plugman/PlugManVelocity.java
@@ -52,7 +52,7 @@ import java.nio.file.Path;
 @Plugin(
         id = "plugmanvelocity",
         name = "PlugManVelocity",
-        version = "3.0.4-dev1",
+        version = "3.0.4",
         description = "Plugin manager for Velocity servers.",
         authors = {"rylinaux", "TestAccount666"}
 )

--- a/plugman-velocity/src/main/java/velocity/com/rylinaux/plugman/PlugManVelocity.java
+++ b/plugman-velocity/src/main/java/velocity/com/rylinaux/plugman/PlugManVelocity.java
@@ -52,7 +52,7 @@ import java.nio.file.Path;
 @Plugin(
         id = "plugmanvelocity",
         name = "PlugManVelocity",
-        version = "3.0.3",
+        version = "3.0.4-dev1",
         description = "Plugin manager for Velocity servers.",
         authors = {"rylinaux", "TestAccount666"}
 )

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
     <groupId>com.rylinaux</groupId>
     <artifactId>PlugManX</artifactId>
-    <version>3.0.3</version>
+    <version>3.0.4-dev1</version>
     <packaging>pom</packaging>
 
     <modules>
@@ -57,8 +57,8 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <plugman.version>3.0.3</plugman.version>
-        <manifold.version>2025.1.28</manifold.version>
+        <plugman.version>3.0.4-dev1</plugman.version>
+        <manifold.version>2025.1.31</manifold.version>
         <jackson.version>2.13.5</jackson.version>
     </properties>
 

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
     <groupId>com.rylinaux</groupId>
     <artifactId>PlugManX</artifactId>
-    <version>3.0.4-dev1</version>
+    <version>3.0.4</version>
     <packaging>pom</packaging>
 
     <modules>
@@ -57,7 +57,7 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <plugman.version>3.0.4-dev1</plugman.version>
+        <plugman.version>3.0.4</plugman.version>
         <manifold.version>2025.1.31</manifold.version>
         <jackson.version>2.13.5</jackson.version>
     </properties>

